lib/packwerk/commands/check_command.rb

```diff
@@ -0,0 +1,60 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Packwerk
+  module Commands
+    class CheckCommand < BaseCommand
+      extend T::Sig
+      include UsesParseRun
+
+      description "run all checks"
+
+      sig { override.returns(T::Boolean) }
+      def run
+        if @files_for_processing.files.empty?
+          @out.puts(<<~MSG.squish)
+            No files found or given.
+            Specify files or check the include and exclude glob in the config file.
+          MSG
+
+          true
+        end
+
+        all_offenses = T.let([], T::Array[Offense])
+        on_interrupt = T.let(-> { @progress_formatter.interrupted }, T.proc.void)
```

Comment:
```suggestion
        on_interrupt = T.let(proc { @progress_formatter.interrupted }, T.proc.void)
```
These could also be memoized private methods